src/main/java/eu/dissco/core/digitalspecimenprocessor/web/FdoRecordBuilder.java

```diff
@@ -0,0 +1,187 @@
+package eu.dissco.core.digitalspecimenprocessor.web;
+
+import static eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileAttributes.REFERENT_NAME;
+import static eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileAttributes.SPECIMEN_HOST;
+import static eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileAttributes.SPECIMEN_HOST_NAME;
+import static eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileAttributes.TOPIC_DISCIPLINE;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import eu.dissco.core.digitalspecimenprocessor.domain.DigitalSpecimen;
+import eu.dissco.core.digitalspecimenprocessor.domain.DigitalSpecimenRecord;
+import eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileAttributes;
+import eu.dissco.core.digitalspecimenprocessor.domain.FdoProfileConstants;
+import eu.dissco.core.digitalspecimenprocessor.exception.PidCreationException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class FdoRecordBuilder {
```

Comment:
As we decided not to implement the Builder pattern I think we should rename it to either FdoRecordComponent or to FdoRecordService, either would be fine to me. Having the builder extension seems to indicate that it follows the pattern.

Reply:
Good point, moved